lib/location/method_gnss.c

```diff
@@ -499,23 +499,10 @@ static void method_gnss_assistance_request(void)
 		method_gnss_nrf_cloud_agps_request();
 #endif
 	}
-#if defined(CONFIG_NRF_CLOUD_PGPS)
-	else if (pgps_agps_request.sv_mask_ephe != 0) {
-		/* When both A-GPS and P-GPS are enabled, this needs to be called only when no
-		 * A-GPS data is requested. When A-GPS data is requested, the nRF Cloud library
-		 * calls the function internally after processing the A-GPS data.
-		 */
-		int err = nrf_cloud_pgps_notify_prediction();
+#endif /* CONFIG_NRF_CLOUD_AGPS */
 
-		if (err) {
-			LOG_ERR("Error requesting notification of prediction availability: %d",
-				err);
-		}
-	}
-#endif /* CONFIG_NRF_CLOUD_PGPS */
-#elif defined(CONFIG_NRF_CLOUD_PGPS)
+#if defined(CONFIG_NRF_CLOUD_PGPS)
 	if (pgps_agps_request.sv_mask_ephe != 0) {
-		/* When only P-GPS is enabled, this needs to be always called. */
```

Comment:
We could have some comment here about nrf_cloud_pgps_notify_prediction call in nrf_cloud library after agps data has been processed and that it doesn't help for the initial pgps request to occur early enough.

Reply:
Added a comment.